frontend/catalyst/jax_primitives.py

```diff
@@ -213,6 +215,34 @@ def _obs_lowering(aval):
 adjoint_p.multiple_results = True
 print_p = jax.core.Primitive("debug_print")
 print_p.multiple_results = True
+python_callback_p = core.Primitive("python_callback")
+python_callback_p.multiple_results = True
+
+
+@python_callback_p.def_abstract_eval
+def _python_callback_abstract_eval(*avals, callback, results_aval):
+    """Abstract evaluation"""
+    return results_aval
+
+
+@python_callback_p.def_impl
+def _python_callback_def_impl(*avals, callback, results_aval):  # pragma: no cover
+    """Concrete evaluation"""
+    raise NotImplementedError()
+
+
+def _python_callback_lowering(jax_ctx: mlir.LoweringRuleContext, *args, callback, results_aval):
+    """Callback lowering"""
+
+    sys.path.append(get_lib_path("runtime", "RUNTIME_LIB_DIR"))
+    import registry  # # pylint: disable=import-outside-toplevel
```

Comment:
It is a bit of a tradeoff. I would like to keep the registry module to be part of the runtime. This is because the name of the library depends on the python version and operating system that is used to compile the Registry.cpp file. `dlopen` needs to know the exact name of the file in order to be able to succeed and it also needs to be available on its search path. The current location is on the search path of the compiled user program, so registry will be able to be found by `dlopen`.

There are two options for having registry be a submodule of catalyst:

1. Move the Registry.cpp file to the frontend and change the build process so that it is done by `setup.py`.
2. We keep the build process in the runtime, just move the generated file to the frontend upon completion.

I don't like option 1 because it is easier to know the name of the function at compile time in the runtime if it is part of the build process of the runtime.

The second option is more palatable. But it involves the runtime's build system being encoded with some knowledge of the frontend's structure and location. Along with the CI/CD scripts and caching. I feel like it is not worthwhile to add the extra complexity here. I am happy to change the name to `catalyst_callback_registry`. If you still feel like the second option is the best way to go, or have other alternatives, happy to listen.

Reply:
Right I see, the fact that is used by both the runtime and frontend complicates this. `catalyst_callback_registry` this is for sure an improvement since `registry` could easily clash with other packages.

Reply:
Renamed here: https://github.com/PennyLaneAI/catalyst/pull/540/commits/7b1717760921f5d7d750b25240f40b4c9fd70056 and here https://github.com/PennyLaneAI/catalyst/pull/540/commits/560a83dbfcbba5eef78bc86273467384f0d7395d . I will rerun the build wheel action one last time to make sure I didn't forget anything in renaming.